src/freenas/usr/local/lib/middlewared_truenas/plugins/failover_/event.py

```diff
@@ -850,6 +854,10 @@ def carp_backup(self, fobj, ifname, vhid, event, user_override):
 
                 self.run_call('etc.generate', 'cron')
 
+                if self.run_call('nis.config')['enable']:
+                    if not self.run_call('nis.started'):
+                        self.run_call('service.restart', 'nis')
```

Comment:
1. do we really need to restart nis when becoming the standby node?
2. if you insist, then we need to add `{'ha_propagate': false}` to this or it will send this request to the active controller